src/pages/api/docker/DockerSingleton.ts

```diff
@@ -9,7 +9,12 @@ export default class DockerSingleton extends Docker {
 
   public static getInstance(): DockerSingleton {
     if (!DockerSingleton.dockerInstance) {
-      DockerSingleton.dockerInstance = new DockerSingleton();
+      DockerSingleton.dockerInstance = new Docker({
+        // If env variable DOCKER_HOST is not set, it will use the default socket
+        ...(process.env.DOCKER_HOST && { host: process.env.DOCKER_HOST }),
```

Comment:
The complexity of this code to read seems quite high.
I would suggest to extract this code to a function.

Reply:
It is not needed. Using the spread operator to conditionally define a property is already done in many different places in the code like in `useGridstack.tsx` or `WrapperContent.tsx`